raven/src/main/java/com/getsentry/raven/RavenFactory.java

```diff
@@ -109,6 +109,20 @@ public static Raven ravenInstance(Dsn dsn, String ravenFactoryName) {
             }
         }
 
+        if (ravenFactoryName != null && triedFactories.size() < 1) {
+            try {
+                // see if the provided class exists on the classpath at all
+                Class.forName(ravenFactoryName);
+                logger.error(
+                    "The raven factory class '{}' is missing a ServiceLoader provider "
+                    + "for com.getsentry.raven.RavenFactory. "
+                    + "See: https://docs.oracle.com/javase/7/docs/api/java/util/ServiceLoader.html", ravenFactoryName);
```

Comment:
- It would probably be more helpful for us to link to our own documentation since the standard library documentation can be a little obtuse.
- This technically could also be due to a missing entry in `MANUALLY_REGISTERED_FACTORIES`, but I'm not sure if that's significant.
- Maybe `s/raven factory/RavenFactory/` here and below, but that's just my opinion.